pycbc/inference/sampler/dynesty.py

```diff
@@ -152,16 +205,84 @@ def from_config(cls, cp, model, output_file=None, nprocesses=1,
         obj = cls(model, nlive=nlive, nprocesses=nprocesses,
                   loglikelihood_function=loglikelihood_function,
                   use_mpi=use_mpi, run_kwds=run_extra, **extra)
-        setup_output(obj, output_file)
+        setup_output(obj, output_file, check_nsamples=False)
+
         if not obj.new_checkpoint:
             obj.resume_from_checkpoint()
         return obj
 
     def checkpoint(self):
-        pass
+        """Checkpoint function for dynesty sampler
+        """
+        with loadfile(self.checkpoint_file, 'a') as fp:
+            fp.write_random_state()
+
+            # Dynesty has its own __getstate__ which deletes
+            # random state information and the pool
+            saved = {}
+            for key in self.no_pickle:
+                if hasattr(self._sampler, key):
+                    saved[key] = getattr(self._sampler, key)
+                    setattr(self._sampler, key, None)
+
+            # For the dynamic sampler, we must also handle the internal
+            # sampler object
+            #saved_internal = {}
+            #if self.nlive < 0:
+            #    for key in self.no_pickle:
+            #        if hasattr(self._sampler.sampler, key):
+            #            saved[key] = getattr(self._sampler.sampler, key)
+            #            setattr(self._sampler.sampler, key, None)
+
+            #for key in self._sampler.__dict__:
+            #    print(key, type(self._sampler.__dict__[key]))
+
+            #for key in self._sampler.sampler.__dict__:
+            #    print(key, type(self._sampler.sampler.__dict__[key]))
```

Comment:
I take the commented out things here will be uncommented once a new release of dynesty is made?

Reply:
Yes, assuming it picks up pr 188